url78d0cbe382_TeeAaTeeUu_SitsiPlaseeraus_tgz-pJ8-sitsiplaseeraus_MainJ8/src/Lataaja/TiedostonKasittelija.java

```diff
@@ -0,0 +1,234 @@
+package Lataaja;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import omatTietorakenteet.ArrayList;
+import omatTietorakenteet.Hakemisto;
+import omatTietorakenteet.Vektori;
+import sitsiplaseeraus.Sitsaaja;
+
+/**
+ * Lukee tiedostosta asetukset, joiden avulla asetusten mukainen sitsi-olio on
+ * mahdollista luoda.
+ */
+public class TiedostonKasittelija {
+
+    private final String file;
+    private Sitsaaja sitsaaja;
+    private Hakemisto<Sitsaaja, Hakemisto> yhteydet;
+    private Hakemisto<String, Sitsaaja> sitsaajat;
+    private String[] tiedot;
+    private Hakemisto<Sitsaaja, Integer> yhteys;
+    private Sitsaaja kohdeSitsaaja;
+    private Hakemisto<Sitsaaja, Integer> valiaikainenMap;
+    private Hakemisto<Sitsaaja, Hakemisto> palautettavaMap;
+    private ArrayList<Sitsaaja> sitsaajatLista;
+    private int poytienMaara;
+    private ArrayList<Integer> poytienKoot;
+    private int arvo;
+    private int index;
+
+    /**
+     * Alustaa ja asettaa tiedoston, josta tiedot otetaan.
+     *
+     * @param tiedosto
+     */
+    public TiedostonKasittelija(String tiedosto) {
+        this.file = tiedosto;
+        this.alustaLuokat();
+    }
+
+    /**
+     * Palauttaa asetustiedostosta löytyneet sitsaajat.
+     *
+     * @return Sitsaajat
+     */
+    public ArrayList<Sitsaaja> getSitsaajat() {
+        sitsaajatLista.clear();
+        for (Vektori<String, Sitsaaja> sitsaaja : sitsaajat) {
+            sitsaajatLista.add(sitsaaja.getValue());
+            if (sitsaaja.getValue().yhteyksienMaara() == 0) {
+                System.out.println("yhteyksiä nolla");
+            }
+        }
+        return sitsaajatLista;
+    }
+
+    /**
+     * Palauttaa pöytien määrän.
+     *
+     * @return Pöytien määrä.
+     */
+    public int getPoytienMaara() {
+        return poytienMaara;
+    }
+
+    /**
+     * Palauttaa pöytien koot taulukossa.
+     *
+     * @return Pöytien koot.
+     */
+    public ArrayList<Integer> getPoytienKoot() {
+        return poytienKoot;
+    }
+
+    /**
+     * Lukee tiedoston ja tallentaa muistiin siitä löytyvät tiedot.
+     *
+     * http://www.mkyong.com/java/how-to-read-and-parse-csv-file-in-java/
+     */
+    public void run() {
+        yhteydet.clear();
+        sitsaajat.clear();
+
+
+        String csvFile = file;
+        BufferedReader br = null;
+        String line = "";
+        String cvsSplitBy = ",";
+        try {
+            br = new BufferedReader(new FileReader(csvFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Currently, there are execution paths that do not contain closure statements, for example, when BufferedReader.readLine (in method avecit) throws an exception. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).